src/session_impl.cpp

```diff
@@ -5068,17 +5062,12 @@ namespace aux {
 		return bind_ep;
 	}
 
-	// verify that the interface ``addr`` belongs to, allows incoming connections
+	// verify that ``addr``s interface allows incoming connections
 	bool session_impl::verify_incoming_interface(address const& addr)
 	{
-		for (auto const& s : m_listen_sockets)
-		{
-			if (s->local_endpoint.address() == addr)
-			{
-				return s->incoming == duplex::accept_incoming;
-			}
-		}
-		return false;
+		return std::any_of(m_listen_sockets.begin(), m_listen_sockets.end()
```

Comment:
this should be a `find_if`

Reply:
why?

Reply:
the `any_of` logic will keep searching, even after if finds `addr`, but the previous loop would terminate immediately after finding the interface with the given address

Reply:
hmm...I find it strange, since I would expect `any_of` to be implemented in terms of `find_if` and avoid the iterator check, but OK (maybe that's not specified).

Reply:
Sorry, disregard my previous comment, now I understand

Reply:
right, but there are two conditions here, and in my first patch (which I just replaced with a force-push) I baked both tests together into one. The current behavior use only one condition for the search, and then the second condition for the return value